src/model/WaterHeaterMixed.cpp

```diff
@@ -1216,6 +1220,123 @@ namespace model {
       }
     }
 
+    boost::optional<PlantLoop> WaterHeaterMixed_Impl::sourceSidePlantLoop() const {
+      if (m_sourceSidePlantLoop) {
+        return m_sourceSidePlantLoop;
+      } else {
+        boost::optional<HVACComponent> sourceSideOutletHVACComponent;
+
+        if (auto t_sourceSideOutletModelObject = sourceSideOutletModelObject()) {
+          sourceSideOutletHVACComponent = t_sourceSideOutletModelObject->optionalCast<HVACComponent>();
+        }
+
+        if (sourceSideOutletHVACComponent) {
+          auto plantLoops = this->model().getConcreteModelObjects<PlantLoop>();
+
+          for (const auto& plantLoop : plantLoops) {
+            if (!plantLoop.supplyComponents(plantLoop.supplyInletNode(), sourceSideOutletHVACComponent.get()).empty()) {
+              m_sourceSidePlantLoop = plantLoop;
+              return plantLoop;
+            }
+          }
+        }
+      }
+
+      return boost::none;
+    }
```

Comment:
Same, I think this should be name alias only. override secondaryPlantLoop, and have it return any plant loop (supply or demand side) connected on the Source side.
```suggestion
    boost::optional<PlantLoop> WaterHeaterMixed_Impl::secondaryPlantLoop() const {
      if (m_secondaryPlantLoop) {
        return m_secondaryPlantLoop;
      } else {

        // Regular case, it's on the demand side: use the base class implementation
        m_secondaryPlantLoop = WaterToWaterComponent_Impl::secondaryPlantLoop();
        if (m_secondaryPlantLoop) {
          return m_secondaryPlantLoop;
        }

        // Less common case: it's also on the supply side (like for the PlantLoop one)
        // so we explicitly ensure this is connected to the **Source** Side Nodes
        if (auto sourceSideOutletModelObject_ = sourceSideOutletModelObject()) {
          if (auto sourceSideOutletHVACComponent_ = sourceSideOutletModelObject_->optionalCast<HVACComponent>()) {
            auto plantLoops = this->model().getConcreteModelObjects<PlantLoop>();

            for (const auto& plantLoop : plantLoops) {
              if (!plantLoop.supplyComponents(plantLoop.supplyInletNode(), sourceSideOutletHVACComponent_.get()).empty()) {
                m_secondaryPlantLoop = plantLoop;
                return plantLoop;
              }
            }
          }
        }
      }

      return boost::none;
    }
```